src/mame/drivers/dfruit.cpp

```diff
@@ -103,95 +74,11 @@ WRITE_LINE_MEMBER(dfruit_state::screen_vblank)
 	}
 }
 
-READ8_MEMBER(dfruit_state::dfruit_rom_r)
-{
-	uint8_t *ROM = memregion("maincpu")->base();
-
-	return ROM[offset + m_rom_bank * 0x2000];
-}
-
-READ8_MEMBER(dfruit_state::dfruit_rom_bank_r)
-{
-	return m_rom_bank;
-}
-
-WRITE8_MEMBER(dfruit_state::dfruit_rom_bank_w)
-{
-	m_rom_bank = data;
-}
-
-READ8_MEMBER(dfruit_state::dfruit_irq_vector_r)
-{
-	return m_irq_vector[offset];
-}
-
-WRITE8_MEMBER(dfruit_state::dfruit_irq_vector_w)
-{
-	m_irq_vector[offset] = data;
-}
-
-READ8_MEMBER(dfruit_state::dfruit_irq_enable_r)
-{
-	return m_irq_enable;
-}
-
-WRITE8_MEMBER(dfruit_state::dfruit_irq_enable_w)
-{
-	m_irq_enable = data;
-}
-
-READ8_MEMBER(dfruit_state::dfruit_ram_bank_r)
-{
-	return m_ram_bank[offset];
-}
-
-WRITE8_MEMBER(dfruit_state::dfruit_ram_bank_w)
-{
-	m_ram_bank[offset] = data;
-}
-
-uint8_t dfruit_state::ram_bank_r(uint16_t offset, uint8_t bank_num)
-{
-	return m_vdp->space().read_byte(offset + (m_ram_bank[bank_num]) * 0x1000);;
-}
-
-void dfruit_state::ram_bank_w(uint16_t offset, uint8_t data, uint8_t bank_num)
-{
-	m_vdp->space().write_byte(offset + (m_ram_bank[bank_num]) * 0x1000, data);;
-}
-
-READ8_MEMBER(dfruit_state::dfruit_ram_0_r) { return ram_bank_r(offset, 0); }
-READ8_MEMBER(dfruit_state::dfruit_ram_1_r) { return ram_bank_r(offset, 1); }
-READ8_MEMBER(dfruit_state::dfruit_ram_2_r) { return ram_bank_r(offset, 2); }
-READ8_MEMBER(dfruit_state::dfruit_ram_3_r) { return ram_bank_r(offset, 3); }
-WRITE8_MEMBER(dfruit_state::dfruit_ram_0_w) { ram_bank_w(offset, data, 0); }
-WRITE8_MEMBER(dfruit_state::dfruit_ram_1_w) { ram_bank_w(offset, data, 1); }
-WRITE8_MEMBER(dfruit_state::dfruit_ram_2_w) { ram_bank_w(offset, data, 2); }
-WRITE8_MEMBER(dfruit_state::dfruit_ram_3_w) { ram_bank_w(offset, data, 3); }
-
-void dfruit_state::tc0091lvc_map(address_map &map)
-{
-	map(0x0000, 0x5fff).rom();
-	map(0x6000, 0x7fff).r(FUNC(dfruit_state::dfruit_rom_r));
-
-	map(0x8000, 0x9fff).ram();
-
-	map(0xc000, 0xcfff).rw(FUNC(dfruit_state::dfruit_ram_0_r), FUNC(dfruit_state::dfruit_ram_0_w));
-	map(0xd000, 0xdfff).rw(FUNC(dfruit_state::dfruit_ram_1_r), FUNC(dfruit_state::dfruit_ram_1_w));
-	map(0xe000, 0xefff).rw(FUNC(dfruit_state::dfruit_ram_2_r), FUNC(dfruit_state::dfruit_ram_2_w));
-	map(0xf000, 0xfdff).rw(FUNC(dfruit_state::dfruit_ram_3_r), FUNC(dfruit_state::dfruit_ram_3_w));
-
-	map(0xfe00, 0xfeff).rw(m_vdp, FUNC(tc0091lvc_device::vregs_r), FUNC(tc0091lvc_device::vregs_w));
-	map(0xff00, 0xff02).rw(FUNC(dfruit_state::dfruit_irq_vector_r), FUNC(dfruit_state::dfruit_irq_vector_w));
-	map(0xff03, 0xff03).rw(FUNC(dfruit_state::dfruit_irq_enable_r), FUNC(dfruit_state::dfruit_irq_enable_w));
-	map(0xff04, 0xff07).rw(FUNC(dfruit_state::dfruit_ram_bank_r), FUNC(dfruit_state::dfruit_ram_bank_w));
-	map(0xff08, 0xff08).rw(FUNC(dfruit_state::dfruit_rom_bank_r), FUNC(dfruit_state::dfruit_rom_bank_w));
```

Comment:
These address maps aided in spotting up the HW for gemcrush, so please reinstate those in both drivers comment out.

Reply:
Just the internal non-common address maps (i.e. anything lingering in the 0xfe00-0xffff area) needs to be moved inside the actual map, also use C++ style commentary. If you want to save some typing, any major IDE supports _comment code block_